Src/java/quick/src/main/resources/org/hl7/fhir/FHIRHelpers-4.0.1.cql

```diff
@@ -107,6 +115,37 @@ define function ToConcept(concept FHIR.CodeableConcept):
             display: concept.text.value
         }
 
+define function reference(reference String):
+    if reference is null then
+        null
+    else
+        Reference { reference: string { value: reference } }
+
+define function resolve(reference String) returns Resource: external
+define function resolve(reference Reference) returns Resource: external
+define function reference(resource Resource) returns Reference: external
+define function extension(element Element, url String) returns List<Element>: external
+define function extension(resource Resource, url String) returns List<Element>: external
+define function hasValue(element Element) returns Boolean: external
+define function getValue(element Element) returns Any: external
+define function ofType(identifier String) returns List<Any>: external
+define function is(identifier String) returns Boolean: external
+define function as(identifier String) returns Any: external
+define function elementDefinition(element Element) returns ElementDefinition: external
+define function slice(element Element, url String, name String) returns List<Element>: external
+define function checkModifiers(resource Resource) returns Resource: external
+define function checkModifiers(resource Resource, modifier String) returns Resource: external
+define function checkModifiers(element Element) returns Element: external
+define function checkModifiers(element Element, modifier String) returns Element: external
+define function conformsTo(resource Resource, structure String) returns Boolean: external
+define function memberOf(code code, valueSet String) returns Boolean: external
+define function memberOf(coding Coding, valueSet String) returns Boolean: external
+define function memberOf(concept CodeableConcept, valueSet String) returns Boolean: external
+define function subsumes(coding Coding, subsumedCoding Coding) returns Boolean: external
+define function subsumes(concept CodeableConcept, subsumedConcept CodeableConcept) returns Boolean: external
+define function subsumedBy(coding Coding, subsumingCoding Coding) returns Boolean: external
+define function subsumedBy(concept CodeableConcept, subsumingConcept CodeableConcept) returns Boolean: external
+define function htmlChecks(element Element) returns Boolean: external
```

Comment:
Yikes.  All of these externals!  In FHIRHelpers!  Can you document the expectations here?  As an engine maintainer do I now need to worry about creating native implementations of all of these functions?  What do they even mean (e.g., `htmlChecks`)? Does the translator implicitly add calls to these?

Reply:
These are the external declarations to support the FHIR-specific FHIRPath extensions:
http://hl7.org/fhir/fhirpath.html
So yes, for an engine to properly support the FHIR-specific FHIRPath extensions, it would need to support these functions. I named them external basically as a place-holder for implementation because they are unlikely to be implemented as raw CQL or FHIRPath, they are engine capabilities that are going to need to be provided. Worst-case the engine would throw an exception run-time (and could advertise with the new capabilities framework that it didn't support those operations).

Reply:
OK.  I recognized some as FHIRPath (e.g., `resolve`), but not all.  As long as those only get called if the library itself calls them, then I'm not too concerned -- but I would be concerned if the translator implicitly adds them to a library that otherwise did not use them...  Are there cases where the translator adds a call like `resolve` in order to follow a reference?

Reply:
No, none of those are implicit, they must be called directly (at this point). We may introduce automatic resolution based on a modelinfo mapping, but that's a next step (profile-informed authoring).